libcloudforensics/providers/kubernetes/volume.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Kubernetes volume class."""
+from typing import Any
+from typing import Dict
+from typing import Optional
+
+from kubernetes import client
+
+
+class K8sVolume:
+  """Class wrapping a Kubernetes volume response."""
+
+  def __init__(self, response: client.V1Volume):
+    """Builds a K8sVolume object.
+
+    Args:
+      response (client.V1Volume): The Kubernetes Volume response object to wrap.
+    """
+    self._response = response
+
+  def Name(self) -> str:
+    """Returns the name of this volume.
+
+    Returns:
+      str: The name of this volume.
+    """
+    name = self._response.name  # type: str
+    return name
+
+  def Type(self) -> str:
+    """Returns the type of this volume.
+
+    Returns:
+      str: The type of this volume.
+
+    Raises:
+      RuntimeError: If the type of this volume is not found.
+    """
+    # There is no attribute for a type, but rather the corresponding type
+    # attribute is non-null.
+    # https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1Volume.md  # pylint: disable=line-too-long
+    response_dict = self._response.to_dict()  # type: Dict[str, Any]
+    for k, v in response_dict.items():
+      if k != 'name' and v:
+        return k
```

Comment:
you return k not v, is this on purpose?

I checked https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1Volume.md it isn't really clear on what is in the response but I see different properties, one is "name" and one is "type", here you assume that if key is not name then the type is in the key itself, is this the case out of testing?

Reply:
Yes it is on purpose

The only other set property (value not being `None`) than "name" seems to be one of these types. There is no "type" property, so I only check `!= 'name'`